ui/src/timeMachine/components/Vis.tsx

```diff
@@ -133,8 +135,15 @@ const mstp = (state: AppState) => {
 
   const giraffeResult = getVisTable(state)
   const xColumn = getXColumnSelection(state)
-  const yColumn = getYColumnSelection(state)
-  const fillColumns = getFillColumnsSelection(state)
+
+  let yColumn = getYColumnSelection(state)
+  let fillColumns = getFillColumnsSelection(state)
+
+  if (activeTimeMachine.view.properties.type === 'mosaic') {
```

Comment:
it would be really nice to do this in the `getYColumnSelection` and `getFillColumnsSelection` functions themselves?

Reply:
let me know if it makes sense?

Reply:
done!